src/executable_path.rs

```diff
@@ -98,4 +112,44 @@ mod canonicalize {
         assert_eq!(file_name, PathBuf::from("ls"));
         Ok(())
     }
+
+    mod executables_that_do_not_exist {
+        use super::*;
+        use pretty_assertions::assert_eq;
+
+        #[test]
+        fn shortens_executables_in_bin_that_do_not_exist() -> R<()> {
+            let executable = Path::new("/bin/doesnotexist");
+            let file_name = canonicalize(executable);
+            assert_eq!(file_name, PathBuf::from("doesnotexist"));
+            Ok(())
+        }
+
+        #[test]
+        fn does_not_shorten_executables_in_subdirectory_of_bin_that_do_not_exist() -> R<()> {
+            let executable = Path::new("/bin/sub/doesnotexist");
+            let file_name = canonicalize(executable);
+            assert_eq!(file_name, PathBuf::from("/bin/sub/doesnotexist"));
+            Ok(())
+        }
+    }
+}
+
+pub fn is_in_bin(executable: &Path) -> bool {
```

Comment:
Rename to something that mentions that this is for pretending that executables exist. `is_fake_executable`? `is_pretend_executable`?

Reply:
How about `should_assume_in_path` ?

Reply:
Yes, thanks!